使用 CODING DevOps 自动部署此项目到境内

## Feature Request

请描述下这个功能的作用：

  1、这个功能的使用场景是什么？

目前此项目使用 GitHub DevOps 整套方案部署到境外，而中国的出国网络不佳，出现丢包甚至无法访问，中国公司往往需要把网站放在境内，以确保境内用户能够快速访问。

DevOps 解决方案 | 代码托管  | 持续集成和部署 | 静态网站架构 | 面向用户
------------------|-----------|----------------|--------------|----------
GitHub | GitHub | GitHub Actions | GitHub Pages | 境外
CODING | CODING | CODING CI（Jenkins） | 云存储+CDN | 境内

  2、这个功能解决你的什么问题？

使用 CODING DevOps 部署此项目到境内，解决境内访问问题，还可供大家学习和了解。

## Other reference

https://help.coding.net/docs/devops/ci/practice/static-website-paas.html

root 的问题解决了，参考 https://github.com/scullyio/scully/issues/194

新的报错：

```
> scully --scanRoutes "--host=0.0.0.0" "--scanRoutes" "--serverTimeout=90000"

Route "/skilltree" rendered into file: "/root/workspace/dist/static/skilltree/index.html"
Wrote 52 routes to sitemap.xml
Route "/helper" rendered into file: "/root/workspace/dist/static/helper/index.html"
Wrote 53 routes to sitemap.xml
Puppeteer error while rendering "/tool" TimeoutError: Navigation timeout of 30000 ms exceeded
Puppeteer error while rendering "/tool" TimeoutError: Navigation timeout of 30000 ms exceeded
npm ERR! code ELIFECYCLE
npm ERR! errno 15
npm ERR! ledge@0.0.0 scully: `scully --scanRoutes "--host=0.0.0.0" "--scanRoutes" "--serverTimeout=90000"`
npm ERR! Exit status 15
npm ERR! 
npm ERR! Failed at the ledge@0.0.0 scully script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

又挂了……@phodal

```
Wrote 78 routes to sitemap.xml
Route "/maturity/msmm" rendered into file: "/root/workspace/dist/static/maturity/msmm/index.html"
Wrote 79 routes to sitemap.xml
Route "/" rendered into file: "/root/workspace/dist/static/index.html"
Wrote 80 routes to sitemap.xml
Puppeteer error while rendering "/tool" TimeoutError: Navigation timeout of 30000 ms exceeded
Puppeteer error while rendering "/tool" TimeoutError: Navigation timeout of 30000 ms exceeded
npm ERR! code ELIFECYCLE
npm ERR! errno 15
npm ERR! ledge@0.0.0 scully: `scully --scanRoutes "--host=0.0.0.0" "--scanRoutes" "--serverTimeout=90000"`
npm ERR! Exit status 15
npm ERR! 
npm ERR! Failed at the ledge@0.0.0 scully script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -1,0 +1,65 @@
+pipeline {
+  agent {
+    docker { 
+      image 'node:12'
+      reuseNode true
+      args '-v /usr/local/share/.cache/yarn/:/usr/local/share/.cache/yarn/'
+    }
+  }
+  stages {
+    stage('检出') {
+      steps {
+        checkout([
+          $class: 'GitSCM',
+          branches: [[name: env.GIT_BUILD_REF]], 
+          userRemoteConfigs: [[url: env.GIT_REPO_URL, credentialsId: env.CREDENTIALS_ID]]
+        ])
+      }
+    }
+    stage('安装依赖') {
+      steps {
+        sh '''find /etc/apt/ -name "*.list" -print0 |
+        xargs -0 sed -i "s/[a-z]\\+.debian.org/mirrors.cloud.tencent.com/g"
+        '''
+        sh 'apt-get update'
+        sh "apt-get install -y gconf-service libasound2 libatk1.0-0 libc6 libcairo2" +
+           " libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4" +
+           " libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0" +
+           " libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1" +
+           " libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6" +
+           " libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation" +
+           " libappindicator1 libnss3 lsb-release xdg-utils wget python3-pip"
+        sh 'yarn install'
+      }
+    }
+    stage('检查代码规范') {
+      steps {
+        sh 'yarn lint'
+      }
+    }
+    stage('构建') {
+      steps {
+        sh 'yarn build:ci'
+      }
+    }
+    stage('测试') {
+      steps {
+        sh 'yarn test:ci'
+      }
+    }
+    stage('部署到腾讯云存储') {
+      when {
+        anyOf {
+          branch 'master'
+          tag '*'
+        }
+      }
+      steps {
+        sh 'pip3 install coscmd'
+        sh "coscmd config -a $COS_SECRET_ID -s $COS_SECRET_KEY" +
+           " -b $COS_BUCKET_NAME -r $COS_BUCKET_REGION"
+        sh 'coscmd upload -r dist/ledge/ /'
+      }
+    }
+  }
+}

--- a/README.md
+++ b/README.md
@@ -6,15 +6,21 @@
 [![codecov](https://codecov.io/gh/phodal/ledge/branch/master/graph/badge.svg)](https://codecov.io/gh/phodal/ledge)
 [![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fphodal%2Fledge.svg?type=shield)](https://app.fossa.com/projects/git%2Bgithub.com%2Fphodal%2Fledge?ref=badge_shield)
 
+[![CODING 持续集成](https://coding-public.coding.net/badges/ledge/job/236475/master/build.svg)](https://coding-public.coding.net/p/ledge/d/ledge/git)
+[![腾讯云 IPv6 解决方案](https://img.shields.io/badge/IPv6-Ready-brightgreen)](https://cloud.tencent.com/solution/ipv6)
+[![SSL 评级](https://img.shields.io/badge/MySSL-A%2B-brightgreen)](https://myssl.com/ledge.devops.host)
+[![HTTP/2](https://img.shields.io/badge/HTTP%2F2-enabled-brightgreen)](https://http2.pro/check?url=https%3A//ledge.devops.host/)
+
 <p align="center"> 
   <img src="src/assets/images/logo.svg" width="288" height="74">
 </p>
 
 在线使用：[https://devops.phodal.com/](https://devops.phodal.com/)
 
-国内服务器（不定期同步）：
+国内服务器：
 
-1. 腾讯云-云开发服务器： [https://ledge.wdsm.io/](https://ledge.wdsm.io/)
+1. CODING（每小时同步）：[https://ledge.devops.host/](https://ledge.devops.host/)
+2. 腾讯云-云开发服务器（不定期同步）： [https://ledge.wdsm.io/](https://ledge.wdsm.io/)
 
 Ledge （from Know-Ledge，意指承载物）知识和工具平台，是我们基于在 ThoughtWorks 进行的一系列 DevOps 实践、敏捷实践、软件开发与测试、精益实践提炼出来的知识体系。它包含了各种最佳实践、原则与模式、实施手册、度量、工具，用于帮助您的企业在数字化时代更好地前进，还有 DevOps 转型。
 
@@ -141,6 +147,5 @@ others see in the Code.
 [![Phodal's Idea](http://brand.phodal.com/shields/idea-small.svg)](http://ideas.phodal.com/)
 
 @ 2020 A [Phodal Huang](https://www.phodal.com)'s [Idea](http://github.com/phodal/ideas). This code is distributed under the MPL license. See `LICENSE` in this directory.
-
 
 [![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fphodal%2Fledge.svg?type=large)](https://app.fossa.com/projects/git%2Bgithub.com%2Fphodal%2Fledge?ref=badge_large)

